roles/openshift_sanitize_inventory/tasks/main.yml

```diff
@@ -19,10 +25,11 @@
     openshift_release: "{{ openshift_release | string | regex_replace('^v', '') }}"
   when: openshift_release is defined
 
-- name: Ensure a valid deployment type has been given.
-  # this variable is required; complain early and clearly if it is invalid.
-  when: openshift_deployment_type not in known_openshift_deployment_types
+- name: Validate openshift_release looks like a version
+  when: |
+    openshift_release is defined
+    and not openshift_release | match('\d+(\.\d+){1,3}$')
```

Comment:
assert doesn't improve things here, it actually forces me to use an or. And the output is more uselessly verbose when it succeeds (which it mostly will).

fail:
```
TASK [openshift_sanitize_inventory : Ensure a valid deployment type has been given] ***
skipping: [localhost]
```

vs assert:
```
TASK [openshift_sanitize_inventory : Ensure a valid deployment type has been given] ***
ok: [localhost] => {
    "changed": false
}
```